analyses/copy_number_consensus_call/src/scripts/get_rid_bad_segments.py

```diff
@@ -0,0 +1,135 @@
+
+################## PURPOSE #################
+# This script is to get rid of unwanted/trouble regions like telomeric, centromeric, and known seg-dups regions
+# This script is to take in 2 files
+#      1st: A reference/list of bad segments that contains the bad segments and their start/end positions
+#      2nd: A list of CNVs that need the bad segments to be filtered out
+
+################# ASSUMPTION ###############
+# It is assumed that the reference file DO NOT have overlapping telomeric/centromeric/segments
+# The provided reference file "bad_chromosomal_seg_updated_merged.txt" DOES NOT have
+#   overlaping segments - thus the assumption is not violated.
+# If the use of a different reference file is desired, make sure to merge the file first before using this script.
+
+############################################
+
+# Imports in the pep8 order https://www.python.org/dev/peps/pep-0008/#imports
+# Standard library
+import argparse
+import subprocess
+import sys
+import os
+
+parser = argparse.ArgumentParser(description="""This script filters out telomeric,
+                                             centromeric, and seg-dup regions""")
+parser.add_argument('--reference', required=True,
+                    help='path to the reference list of bad segments to be filtered out for')
+parser.add_argument('--file', required=True,
+                    help='path to the CNVs file that needs to be filtered')
+
+args = parser.parse_args()
+
+#querie_file = sys.argv[2]
+
+#db_file = sys.argv[1]    #bad segment list
+
+
+## Check to see if the CNVs file is empty, if so, make an empty variable
+if os.stat(args.file).st_size == 0:
+    final_content = []
+
+## If the file has content, then load in the content
+else:
+    with open (args.file) as file:
+
+        ## Strip the trailing spaces for each line
+        stripped_content = [i.rstrip() for i in file.readlines()]
+
+        ## If the file is TAB delimited, split the file up by TABs
+        if stripped_content[0].find('\t') != -1:
+            final_content = [i.split('\t') for i in stripped_content]
+
+        ## If the file is space delimited, split the file up by the spaces
+        else:
+            final_content = [i.split() for i in stripped_content]
```

Comment:
For these purposes, are we likely to get both tab and space separated files? I would have assumed that if these are bed files they should all be separated the same way.

Reply:
For the purpose of this pipeline, the input file into this step is **ALWAYS** going to be space separated files. The reason why this is here is because when I made this script, I wanted to add a little flexibility into it in case someone wants to take this script out and use it on their tab separated data. I could take it out if you think this is not necessary.

Reply:
Would it be safe to use `re.split` and split on any whitespace of one or more characters with `'\s+'`?

Reply:
Also - don't bed files require tab spacing? Are there tools for bed files that output space separated ones?

Reply:
@fingerfen : if `re.split` would let you simplify many lines to one line and reduce the potential for an error, I'm a fan of that.